src/screens/WalletDetailsScreen.tsx

```diff
@@ -131,23 +143,21 @@ export class WalletDetailsScreen extends React.PureComponent<Props> {
   };
 
   onSubscribeButtonPress = () => {
-    const { storedAddress, isWalletSubscribed = true, navigation, wallet } = this.props;
-    if (!isWalletSubscribed) {
-      if (!storedAddress) {
-        // add email and subscribe only certain wallet
+    const { email, navigation, wallet, isSubscribed } = this.props;
+    if (!isSubscribed) {
+      if (!email) {
         return navigation.navigate(Route.Notifications, {
-          walletToSubscribe: wallet,
+          walletsToSubscribe: [wallet!],
         });
       }
       return navigation.navigate(Route.ConfirmEmail, {
-        // subscribe by confirming code
-        address: storedAddress!,
+        address: email!,
```

Comment:
shouldn't be it named email instead of address?

why is the `!` needed?  from what I read it is a little bit of cheating TS and telling the compiler, on my responsibility, it won't be null. I think the better way would be to check explicitly if email === null somewhere above in the function